File: deob/src/main/java/dev/openrs2/deob/remap/PrefixRemapper.java
package dev.openrs2.deob.remap;

import java.util.HashMap;

import dev.openrs2.asm.classpath.Library;
import org.objectweb.asm.commons.Remapper;
import org.objectweb.asm.commons.SimpleRemapper;

public final class PrefixRemapper {
	public static Remapper create(Library library, String prefix) {
		var mapping = new HashMap<String, String>();
		for (var clazz : library) {
			if (TypedRemapper.EXCLUDED_CLASSES.contains(clazz.name)) {
				mapping.put(clazz.name, clazz.name);
			} else {
				mapping.put(clazz.name, prefix + clazz.name);
			}
		}

		return new SimpleRemapper(mapping);
	}

	private PrefixRemapper() {
		/* empty */
	}
}

File: deob/src/main/java/dev/openrs2/deob/remap/PrefixRemapper.kt
package dev.openrs2.deob.remap

import dev.openrs2.asm.classpath.ExtendedRemapper
import dev.openrs2.asm.classpath.Library
import dev.openrs2.asm.filter.ClassFilter

class PrefixRemapper(private val prefix: String, private val classes: Set<String>) : ExtendedRemapper() {
    override fun map(internalName: String): String {
        return if (classes.contains(internalName)) {
            prefix + internalName
        } else {
            internalName
        }
    }

    companion object {
        fun create(library: Library, prefix: String, excluded: ClassFilter): ExtendedRemapper {
            val classes = mutableSetOf<String>()

            for (clazz in library) {
                if (!excluded.matches(clazz.name)) {
                    classes += clazz.name
                }
            }

            return PrefixRemapper(prefix, classes)
        }
    }
}
